samples/bluetooth/direct_adv/src/main.c

```diff
@@ -76,6 +76,7 @@ BT_GATT_SERVICE_DEFINE(primary_service,
 static const struct bt_data ad[] = {
 	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
 	BT_DATA_BYTES(BT_DATA_UUID128_ALL, BT_UUID_CUSTOM_SERVICE_VAL),
+	BT_DATA(BT_DATA_NAME_COMPLETE, CONFIG_BT_DEVICE_NAME, sizeof(CONFIG_BT_DEVICE_NAME) - 1),
```

Comment:
Would it make sense to do
```suggestion
	BT_DATA(BT_DATA_NAME_COMPLETE, CONFIG_BT_DEVICE_NAME, strlen(CONFIG_BT_DEVICE_NAME)),
```

Reply:
Not sure if too specific, but we could considering introducing a `BT_DATA_NAME` macro that does this, so it would just be 

```c

static const struct bt_data ad[] = {
	BT_DATA_NAME,
};
```

Reply:
I prefer to keep the explicit definition, we are trying to remove magic, let's not hide thing again